File: src/config/cors-origins.ts
export function corsOrigins(env: NodeJS.ProcessEnv = process.env): string[] {
  const configured = (env.CORS_ORIGINS || '').split(',').map(origin => origin.trim()).filter(Boolean);
  if (env.NODE_ENV === 'production') return configured;
  // Vite advances to the next port when another local instance is running.
  const local = ['localhost', '127.0.0.1', '[::1]'].flatMap(host =>
    [4173, 5173, 5174, 5175, 5176, 5177, 5178, 5179].map(port => `http://${host}:${port}`),
  );
  return [...new Set([...configured, ...local])];
}

File: analytics-service/app/cors_origins.py
import os


def cors_origins(env=None):
    env = os.environ if env is None else env
    configured = [origin.strip() for origin in env.get("CORS_ORIGINS", "").split(",") if origin.strip()]
    if env.get("NODE_ENV") == "production":
        return configured
    local = [f"http://{host}:{port}" for host in ("localhost", "127.0.0.1", "[::1]")
             for port in (4173, 5173, 5174, 5175, 5176, 5177, 5178, 5179)]
    return list(dict.fromkeys(configured + local))
